pkg/controller/vsphere/reconciler_test.go

```diff
@@ -98,128 +99,230 @@ func TestClone(t *testing.T) {
 	}
 
 	testCases := []struct {
-		testCase    string
-		machine     func(t *testing.T) *machinev1.Machine
-		expectError bool
-		cloneVM     bool
+		testCase              string
+		cloneVM               bool
+		expectedError         error
+		setupFailureCondition func() error
+		providerSpec          vsphereapi.VSphereMachineProviderSpec
+		machineName           string
 	}{
 		{
 			testCase: "clone machine from default values",
-			machine: func(t *testing.T) *machinev1.Machine {
-				providerSpec := vsphereapi.VSphereMachineProviderSpec{
-					CredentialsSecret: &corev1.LocalObjectReference{
-						Name: "test",
-					},
-					Workspace: &vsphereapi.Workspace{
-						Server: server.URL.Host,
-					},
+			providerSpec: vsphereapi.VSphereMachineProviderSpec{
+				CredentialsSecret: &corev1.LocalObjectReference{
+					Name: "test",
+				},
+				Workspace: &vsphereapi.Workspace{
+					Server: server.URL.Host,
+				},
+			},
+			cloneVM:     true,
+			machineName: "test0",
+		},
+		{
+			testCase: "clone machine in specific folder",
+			providerSpec: vsphereapi.VSphereMachineProviderSpec{
+				CredentialsSecret: &corev1.LocalObjectReference{
+					Name: "test",
+				},
+				Workspace: &vsphereapi.Workspace{
+					Server: server.URL.Host,
+					Folder: "custom-folder",
+				},
+			},
+			cloneVM:     true,
+			machineName: "test1",
+		},
+		{
+			testCase: "fail on invalid resource pool",
+			providerSpec: vsphereapi.VSphereMachineProviderSpec{
+				CredentialsSecret: &corev1.LocalObjectReference{
+					Name: "test",
+				},
+				Workspace: &vsphereapi.Workspace{
+					Server:       server.URL.Host,
+					ResourcePool: "invalid",
+				},
+			},
+			expectedError: errors.New("resource pool not found, specify valid value"),
+		},
+		{
+			testCase: "fail on multiple resource pools",
+			providerSpec: vsphereapi.VSphereMachineProviderSpec{
+				CredentialsSecret: &corev1.LocalObjectReference{
+					Name: "test",
+				},
+				Workspace: &vsphereapi.Workspace{
+					Server:       server.URL.Host,
+					ResourcePool: "/DC0/host/DC0_C0/Resources/...",
+				},
+			},
+			expectedError: errors.New("multiple resource pools found, specify one in config"),
+			setupFailureCondition: func() error {
+				// Create resource pools
+				defaultResourcePool, err := session.Finder.ResourcePool(context.Background(), "/DC0/host/DC0_C0/Resources")
```

Comment:
is this magic value `"/DC0/host/DC0_C0/Resources"` completely arbitrary? may be add a comment explaining where is it coming from?

Reply:
That's not a magic value, just the default value that 'simulator' has set. I had to debug this a bit.